component/main.jsonnet

```diff
@@ -17,34 +17,30 @@ local admin_secret = kube.Secret(params.admin.secretname) {
   },
 };
 
-local external_db_secret =
-  local isdummysecret =
-    if params.database.builtin then
-      {
-        'commodore.syn.tools/dummy-secret': 'true',
-      }
-    else
-      {};
-  kube.Secret(params.database.external.secretname) {
-    metadata+: {
-      labels+: params.labels + isdummysecret,
-    },
-    stringData:
-      if !params.database.builtin then
-        {
-          DB_VENDOR: params.database.external.vendor,
-          DB_ADDR: params.database.external.host,
-          DB_PORT: params.database.external.port,
-          DB_DATABASE: params.database.external.database,
-          DB_USER: params.database.external.username,
-          DB_PASSWORD: params.database.external.password,
-        }
-      else {},
-  };
+local secrets = {
+  builtin: {
+    'postgresql-password': params.database.password,
+  },
+  external: {
+    DB_DATABASE: params.database.database,
+    DB_USER: params.database.username,
+    DB_PASSWORD: params.database.password,
+    DB_VENDOR: params.database.external.vendor,
+    DB_ADDR: params.database.external.host,
+    DB_PORT: params.database.external.port,
+  },
+};
+
+local db_secret = kube.Secret(params.database.secretname) {
+  metadata+: {
+    labels+: params.labels,
+  },
+  stringData: secrets[params.database.provider],
+};
 
 // Define outputs below
 {
   '00_namespace': namespace,
```

Comment:
Some questions here:
* How do you want to handle namespacing of instances?
* Should all instances always be in a single namespace? 
* Should the user be able to decide whether instances share a namespace?

At a minimum, the documentation needs to be clear that all instances will be deployed in namespace `syn-keycloak` if the user doesn't override the namespace parameter per instance.

Reply:
I don't know :)
What is the desired convention when developing multi-instance components? from [SDD 25](https://syn.tools/syn/SDDs/0025-commodore-component-instantiation.html) I'm not quite sure about the implications in this case:

>Commodore won’t modify the parameters.<component-name>.namespace field for instantiated components. Component authors need to explicitly implement a namespace per instance, if they so desire.

I would propose to default the namespace to `${_instance}`, but add a line in the migration guide to set it explicitly to `syn-keycloak`. Still, resources should contain the instance name in their names, in case one decides to run multiple instances in the same namespace.

Reply:
There's no convention really, for example, for nfs-subdir-external-provisioner which consists of a single pod per instance (+/-) it makes sense to keep all of them in a single namespace.

For more complex applications or int/prod separation by instance it may be desirable to have the instances in separate namespaces. I think going with `namespace: syn-${_instance}` or `namespace: ${_instance}` as the default would work well, users can still adjust the namespace per instance as they desire.